packages/smart-vaults/src/vault/SmartVault.sol

```diff
@@ -35,13 +35,30 @@ contract SmartVault is IAccount, Ownable, UUPSUpgradeable, MultiSignerAuth, ERC1
         ERC1271
     }
 
+    /// @notice Upper limits for maxPriorityFeePerGas, preVerificationGas, verificationGasLimit and callGasLimit that
+    /// should be charged by the userOp. This is included in the light userOp hash to ensure last signer does not exceed
+    /// the specified gas price/limits. These values will be ignored when threshold is 1.
+    struct LightUserOpGasLimits {
+        uint256 maxPriorityFeePerGas;
+        uint256 preVerificationGas;
+        uint256 callGasLimit;
+        uint256 verificationGasLimit;
```

Comment:
do we have to check either of the paymaster values?
are we sure it isn't easier to check against a sum?

Reply:
still not sure how to do a limit on paymaster because its in paymasters control and we don't know their verification gas until the end?
maybe we can have light signers only sign over paymaster address so no random paymaster can be added by last signer and that should be enough?

Reply:
possibly..

i assume there's an rpc method to get the paymaster gas values and their signature. we could call it, 2x the values they give, and then let it expire (not so different from just estimating gas regularly i think?)

we could also just hardcode reasonable limits on our FE

Reply:
can talk to riley about the different options here monday

Reply:
> i assume there's an rpc method to get the paymaster gas values and their signature.

Yes, see [`eth_estimateUserOperationGas`](https://eips.ethereum.org/EIPS/eip-4337#-eth_estimateuseroperationgas). You would have to pass in fake values in many of these fields and estimate the UserOp. 

It's probably worth testing this somehow soonish, because the UserOps are very complex here.